lib/LMSManagers/LMSUserManager.php

```diff
@@ -119,7 +119,7 @@ public function getUserNames($params = array())
             (CASE WHEN access = 1 AND accessfrom <= ?NOW? AND (accessto >=?NOW? OR accessto = 0) THEN 1 ELSE 0 END) AS access
             FROM vusers
             WHERE deleted = 0'
-            . (isset($withDeleted) ? ' OR deleted = 1' : '' )
+            . (!empty($withDeleted) ? ' OR deleted = 1' : '' )
             . ' ORDER BY rname ASC'
```

Comment:
@rafalpietraszewicz ma to sens?

Reply:
Ja przyjąłem założenie ` (isset($withDeleted) ? ' OR deleted = 1' : '' )`, że aby uwzględnić usuniętych wystarczy, że przekazany zostanie parametr `$withDeleted`, nie ważne z jaką wartością (może być 'true', 1, 0, 'yes'), ważne aby był podany. I to działa. Natomiast@inerduo proponuje,  żeby zamiast isset dać '!empty' bo, według niego, jak ktoś przestawi na 0, to wtedy pokaże usuniętych. Zgadzam się z tym. Tutaj różnica sprowadza się do interpretacji intencji i odpowiedzi na pytanie, czy aby wyświetlić usuniętych wystarczy przekazać  do metody parametr, czy parametr w wartością różną od '0'. Myślę, że przed ewentualnym wprowadzeniem zmiany, @interduo powinien jeszcze sprawdzić jak jest definiowany parametr `$withDeleted` w pozostałym kodzie i czy przypadkiem nie będzie ta zmiana w kolizji z dotychczasowym użyciem parametru.

Reply:
Ja rozumiem jaka idea Ci przyświecała dla mnie ten kod wyglądał początkowo na niedokończony - brakuje również komentarza nad metodą co do obsługi parametru bo blok nad metodą jest.

Reasumując:
- dodajemy komentarz do parametru metody,
- dorabiamy obsługę wartości w parametrze a nie tylko jego obecności/nieobecności,
- ustawiamy wartość domyślną parametru withDeleted=0 by zachować zgodność wstecz,

Będzie git?

Reply:
To już jak @chilek uważa.

Nie wiem, czy się nie mylę, ale jeśli będzie warunek `(!empty($withDeleted) ? ' OR deleted = 1' : '' )` to dla zgodności przekazywany parametr `$withDeleted` powinien mieć wartość '1', a nie '0' (żeby uwzględnić usuniętych użytkowników).

Reply:
Nie zmieniłem domyślnego zachowania (pokazuj tylko **nieusuniętych** użytkowników). Przejrzałem kod pod kątem wystąpienia użycia parametru withDeleted = 0 i withDeleted = null. Nie znajduję błędów z tym.

Reply:
Mnie się ta zmiana nie podoba, bo niby jest rozbudowa działania, ale nadal nie obejmuje wszystkich warunków:
1. Weź użytkowników skasowanych.
2. Weź użytkowników nieskasowanych.
3. Ignoruj zupełnie flagę deleted.

Reply:
Czyli lepiej standardowo jak mamy wszędzie $deleted:
0 (nie skasowani, domyślny wybór)
1 (skasowani)
-1 (wszyscy)
?

Reply:
Pewnie tak.

Reply:
https://github.com/lmsgit/lms/commit/c25126c2c53d46e63fea68460578f12a5d0e9751